articles/quickstart/backend/java-spring-security5/01-authorization.md

```diff
@@ -0,0 +1,317 @@
+* `GET /api/private`: available for authenticated requests containing a valid JWT Access Token
+* `GET /api/private-scoped`: available for authenticated requests containing a valid JWT Access Token with the `read:messages` scope granted
+
+The example below shows how to secure API methods using the `HttpSecurity` object provided in the `configure()` method of the `SecurityConfig` class. Route matchers are used to restrict access based on the level of authorization required:
+
+```java
+// src/main/java/com/auth0/example/security/SecurityConfig.java
+
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Override
+    public void configure(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                .mvcMatchers("/api/public").permitAll()
+                .mvcMatchers("/api/private").authenticated()
+                .mvcMatchers("api/private-scoped").hasAuthority("SCOPE_read:messages")
+                .and()
+                .oauth2ResourceServer().jwt();
+    }
+}
+```
+
+::: note
+By default, Spring Security will create a `GrantedAuthority` for each scope in the `scope` claim of the JWT. This is what enables using the `hasAuthority("SCOPE_read:messages")` method to restrict access to a valid JWT that contains the `read:messages` scope.
+
+If your use case requires different claims to make authorization decisions, see the [Spring Security Reference Documentation](https://docs.spring.io/spring-security/site/docs/current/reference/htmlsingle/#oauth2resourceserver-authorization-extraction) for detailed instructions.
+:::
+
+### Create the API controller
+
+Create a new class named `Message`, which is the domain object the API will return:
+
+```java
+// src/main/java/com/auth0/example/model/Message.java
+
+public class Message {
+    private final String message;
+
+    Message(String message) {
+        this.message = message;
+    }
+
+    public String getMessage() {
+        return this.message;
+    }
+}
+```
+
+Create a new class named `APIController` to handle requests to the endpoints:
+
+```java
+// src/main/java/com/auth0/example/web/APIController.java
+
+@RestController
+@RequestMapping(path = "api", produces = MediaType.APPLICATION_JSON_VALUE)
+public class APIController {
+
+    @GetMapping(value = "/public")
+    public Message publicEndpoint() {
+        return new Message("All good. You DO NOT need to be authenticated to call /api/public.");
+    }
+
+    @GetMapping(value = "/private")
+    public Message privateEndpoint() {
+        return new Message("All good. You can see this because you are Authenticated.");
+    }
+
+    @GetMapping(value = "/private-scoped")
+    public Message privateScopedEndpoint() {
+        return new Message("All good. You can see this because you are Authenticated with a Token granted the 'read:messages' scope");
+    }
+}
+```
+
+## Run and Test Your API
+
+To build and run the sample project, run the `bootRun` Gradle task.
```

Comment:
Given that you mentioned Maven at the start, I'd also explain how to run it with maven, if that's possible for this sample and is no longer than a line or two.